static/panes/editor.js

```diff
@@ -1029,7 +1041,46 @@ Editor.prototype.onSelectLine = function (id, lineNum) {
     }
 };
 
-Editor.prototype.onEditorLinkLine = function (editorId, lineNum, columnNum, reveal) {
+// Returns a half-segment [a, b) for the token on the line lineNum
+// that spans across the column.
+// a - colStart points to the first character of the token
+// b - colEnd points to the character immediately following the token
+// e.g.: "this->callableMethod ( x, y );"
+//              ^a   ^column  ^b
+Editor.prototype.getTokenSpan = function (lineNum, column) {
+    var model = this.editor.getModel();
+    var line = model.getLineContent(lineNum);
+    if (0 < column && column < line.length) {
+        var tokens = monaco.editor.tokenize(line, model.getModeId());
```

Comment:
It's so unfortunate we have to do this. I have another project which needed the same sad piece of code. Thanks for taking the time to work out how this is done!